__tests__/app.test.js

```diff
@@ -111,51 +111,133 @@ describe("/api/articles/:article_id", () => {
+      return request(app)
+        .get("/api/articles/2/comments")
+        .expect(404)
+        .then(({ body }) => {
+          const { msg } = body;
+          expect(msg).toBe("Not found");
+        });
+    });
   });
-  test("GET 404: article_id comments not found", () => {
-    return request(app)
-      .get("/api/articles/2/comments")
-      .expect(404)
-      .then(({ body }) => {
-        const { msg } = body;
-        expect(msg).toBe("Not found");
-      });
+  describe("POST", () => {
+    test("POST 201: Accepts an object with a username and body, and responds with a posted comment", () => {
+      const testComment = {
+        username: "butter_bridge",
+        body: "This is a test comment",
+      };
+      return request(app)
+        .post("/api/articles/2/comments")
+        .send(testComment)
+        .expect(201)
+        .then(({ body }) => {
+          const { comment } = body;
+          const expectedComment = {
+            comment_id: 19,
+            body: "This is a test comment",
+            votes: 0,
+            author: "butter_bridge",
+            article_id: 2,
+            created_at: expect.any(String),
+          };
+          expect(comment).toMatchObject(expectedComment);
+        });
+    });
+    test("POST 400: Invalid article_id type received", () => {
+      const testComment = {
+        username: "butter_bridge",
+        body: "This is a test comment",
+      };
+      return request(app)
+        .post("/api/articles/invalid-input/comments")
+        .send(testComment)
+        .expect(400)
+        .then(({ body }) => {
+          const { msg } = body;
+          expect(msg).toBe("Bad request");
+        });
+    });
+    test("POST 404: article_id not found", () => {
+      const testComment = {
+        username: "butter_bridge",
+        body: "This is a test comment",
+      };
+      return request(app)
+        .post("/api/articles/100/comments")
+        .send(testComment)
+        .expect(404)
+        .then(({ body }) => {
+          const { msg } = body;
+          expect(msg).toBe("Not found");
+        });
+    });
+    test("POST 400: Invalid comment request received", () => {
+      const testComment = {
+        invalid: "butter_bridge",
+        body: "This is a test comment",
+      };
+      return request(app)
+        .post("/api/articles/1/comments")
+        .send(testComment)
+        .expect(400)
+        .then(({ body }) => {
+          const { msg } = body;
+          expect(msg).toBe("Bad request");
+        });
+    });
+    test("POST 404: username not found", () => {
+      const testComment = {
+        username: "invalid",
+        body: "This is a test comment",
+      };
+      return request(app)
+        .post("/api/articles/1/comments")
+        .send(testComment)
+        .expect(404)
+        .then(({ body }) => {
+          const { msg } = body;
+          expect(msg).toBe("Not found");
+        });
```

Comment:
Nicely handled on this post. Might be worth updating your endpoints.json to remind yourself in three weeks time you can only post a comment as a user that exists in the data base. (So many people forget)